### PR TITLE
add frontend utility functions

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: n/a

ECT PR: https://github.com/moderntribe/events-community-tickets/pull/93
ET+ PR: https://github.com/moderntribe/event-tickets-plus/pull/474

### CODE REVIEW

**Comment by tr1b0t** [comment]
Closing this Pull Request because it is pull requested against the wrong branch. The Central ticket is for *M17.24*, but this was pull requested against *release/M17.25*.

File: src/functions/template-tags/general.php [hunk 1]
```diff
@@ -601,3 +601,33 @@ function tribe_assets( $origin, $assets, $action = null, $arguments = array() )
...
+	 */
+	function tribe_doing_frontend( $doing_frontend ) {
+		$callback = $doing_frontend ? '__return_true' : '__return_false';
+
+		add_filter( 'tribe_doing_frontend', $callback );
+	}
+}
+
+if ( ! function_exists( 'tribe_is_frontend' ) ) {
+	function tribe_is_frontend() {
```
Comment (bordoni) [inline src/functions/template-tags/general.php]: Docblock.

Comment (lucatume) [reply src/functions/template-tags/general.php]: right, added typecast too

File: src/functions/template-tags/general.php [hunk 2]
```diff
@@ -601,3 +601,40 @@ function tribe_assets( $origin, $assets, $action = null, $arguments = array() )
...
+	 *
+	 * @since TBd
+	 *
+	 * @param bool $doing_frontend Whether what is being done happens in the
+	 *                             context of the frontend or not.
+	 */
+	function tribe_doing_frontend( $doing_frontend ) {
+		$callback = $doing_frontend ? '__return_true' : '__return_false';
+
+		add_filter( 'tribe_doing_frontend', $callback );
```
Comment (ggwicz) [inline src/functions/template-tags/general.php]: docblock this filter

Comment (lucatume) [reply src/functions/template-tags/general.php]: here we are adding a filter, not applying it.

**Comment by lucatume** [comment]
Added the changelog entry.

**Pull request merged**